### ISSUE TITLE
Markdown previewing for code blocks breaks when 1st line is square bracket

### ISSUE THREAD

**Issue by lwsrbrts** [Microsoft/vscode#1668]
A picture speaks a thousand words. First screenshot, Markdown previewing is incorrect and ordered numbering is subsequently broken also as a result of the first line of the code starting with an open square bracket. When the open square bracket is replaced with another character (a number 1 in the second screenshot), the preview works fine.
![markdown preview with square bracket](https://cloud.githubusercontent.com/assets/15855642/12009586/bbef16a2-ac76-11e5-8858-3833369d4871.png)
![markdown preview with another char](https://cloud.githubusercontent.com/assets/15855642/12009588/c0483c2e-ac76-11e5-8cda-846bf9f1022c.png)

**Comment by bpasero** [comment]
@lwsrbrts can you attach the markdown?

**Comment by lwsrbrts** [comment]
Sorry, that would have been the intelligent thing to do in the first place. :)

https://raw.githubusercontent.com/lwsrbrts/PoSHue/master/README.md

**Comment by bpasero** [comment]
Thanks!

**Comment by golf1052** [comment]
So I did some sleuthing to try to figure out what was going on.  
TL;DR this seems to be an issue in [marked](https://github.com/chjj/marked), not in VSCode.

The results of my tests are [here](https://github.com/golf1052/markdown-testing)  
I tried three different markdown parsers, [John Gruber's implementation](https://daringfireball.net/projects/markdown/), [GitHub's markup](https://github.com/github/markup), and [marked](https://github.com/chjj/marked) (v0.3.5 where VSCode is using v0.3.2)

README.md is your original file cut down to the part that was breaking. README-mod.md removes a space your first powershell code block. Tildes represent one whitespace character (space).
```markdown
~2. Run
~
```powershell
~[HueBridge]::FindHueBridge()
~```
~3.
```
README-1.md adds the one that you added when you were testing.  
.html files are the outputs from each parser. -mod.html indicates that file was generated from README-mod.md and -1.html indicates that file was generated from README-1.md. ref.html files were generated by John Gruber's implementation, github.html files were generated by GitHub's markup, and marked.html files were generated by marked.

None of the 3 parse README.md correctly. Only marked parses README-mod.md correctly. All 3 parse README-1.md correctly. (When I mean parse correctly `[HueBridge]::FindHueBridge()` appears after `2. Run`. Online GitHub parses all 3 files correctly. I also tried [CommonMark's implementation](http://spec.commonmark.org/dingus/) since I guess Gruber's implementation isn't really the reference implementation and it also parses all 3 files correctly.

¯\_(ツ)_/¯

**Comment by octref** [comment]
@bpasero 

[marked](https://github.com/chjj/marked) seems to be dead https://github.com/chjj/marked/issues/724

Maybe switch to a more well-maintained markdown library like https://markdown-it.github.io/ ?
Otherwise this issue can't be fixed.

**Comment by Treri** [comment]
@octref subscribe #4668 please

**Comment by lwsrbrts** [comment]
I've since updated the linked readme.md so it no longer suffers from the issue but the following link is to the point in the repo where I reported the bug.

https://raw.githubusercontent.com/lwsrbrts/PoSHue/4718c4f933031d8c2c8bcdc2bf0561ebe01e067a/README.md

**Comment by bpasero** [comment]
Even with our new markdown extension I still see issues rendering the bullets:

![image](https://cloud.githubusercontent.com/assets/900690/16124272/93f6c604-33ee-11e6-8f17-5f8a560dd808.png)

**Comment by mjbvz** [comment]
Since this issue was originally opened to track the markdown preview problems and those seem resolved, I'm going to close this. 

I've gone through the example file and opened issues to track some of the editor colorization problems:

#12946 - No/bad colorization of codeblock immediately after a paragraph.
#12946 - No colorization of numbered list punctuator immediately after a paragraph.
#3746 - Fenced code blocks language highlighting can break out of blocks
#12948 - No colorization of block quote punctuator if the quote is indented

Please let me know if you are still seeing issues with the preview, and feel free to open additional bugs to track colorization problems in the markdown editor.

Thanks.
